### PR TITLE
Link to Astro docs for heading anchor link generation

### PR DESCRIPTION
**Pull request by onurusluca**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description
It may be difficult to understand how to correctly set anchor links for some people.

Update authoring-content.md about automatic anchoring to clarify that automatic anchor links only accepts lowecase anchors with no space.
 
 Added:

```
:::note
**Convert to Lowercase:** Even if your title has uppercase characters, write them in the anchor link with lowercase. `Title` > `title`
<br/>
**No Spaces in Titles:** If your title has spaces, replace spaces with hyphens(-). `My Title` > `my-title`
:::
```

### CODE REVIEW

**Review by delucis — COMMENTED** [review]
Thank you for the PR @onurusluca! The heading ID generator does a few more things — it is based on https://github.com/Flet/github-slugger. I also wonder whether it would be better to document this on the [Astro docs Markdown page](https://docs.astro.build/en/guides/markdown-content/#heading-ids) and link to it? We just use Astro’s default anchor link behaviour, so it’s not Starlight specific. What do you think?

**Comment by onurusluca** [comment]
@delucist Thanks for the reply. 
I see, I didn't know Starlight used Astro's anchor link that is based on github slugger.

We can definitely link to the page. Let me fix the commit.
Thanks for asking btw.

**Comment by onurusluca** [comment]
@delucis Is this better maybe? You are welcome to change it. 


:::note
**Convert to Lowercase:** Even if your title has uppercase characters, write them in the anchor link with lowercase. `Title` > `title`

**No Spaces in Titles:** If your title has spaces, replace spaces with hyphens(-).
`My Title` > `my-title`

Starlight uses the same [Heading IDS logic](https://github.com/Flet/github-slugger?tab=readme-ov-file)
as Astro which is based on [ github-slugger ](https://github.com/Flet/github-slugger?tab=readme-ov-file).
:::

**Comment by ElianCodes** [comment]
Hey there @onurusluca! I've made a PR to the Astro docs itself. I feel like that's a better place to document this, since it's the same behaviour for both Starlight and Astro.

We can link to that page from Starlight. How does that sound?

PR here: https://github.com/withastro/docs/pull/5300

File: docs/src/content/docs/guides/authoring-content.md [hunk 1]
```diff
@@ -89,6 +89,16 @@ I can link to [my conclusion](#conclusion) lower on the same page.
...
+:::note
+**Convert to Lowercase:** Even if your title has uppercase characters, write them in the anchor link with lowercase. `Title` > `title`
+
+**No Spaces in Titles:** If your title has spaces, replace spaces with hyphens(-).
+`My Title` > `my-title`
+
+Starlight uses the same [Heading IDS logic](https://github.com/Flet/github-slugger?tab=readme-ov-file)
+as Astro which is based on [ github-slugger ](https://github.com/Flet/github-slugger?tab=readme-ov-file).
+:::
+
```
Comment (ElianCodes) [inline docs/src/content/docs/guides/authoring-content.md:101]:
```suggestion
Learn more about how Astro processes heading `id`s in [the Astro Documentation](https://docs.astro.build/en/guides/markdown-content/#heading-ids)
```

**Review by ElianCodes — COMMENTED** [review]
Since I added this piece of documentation to the main Astro Docs, we can link to that instead of writing it out here.

**Comment by onurusluca** [comment]
> Hey there @onurusluca! I've made a PR to the Astro docs itself. I feel like that's a better place to document this, since it's the same behaviour for both Starlight and Astro.
> 
> We can link to that page from Starlight. How does that sound?
> 
> PR here: [withastro/docs#5300](https://github.com/withastro/docs/pull/5300)

Thanks for the PR. That looks fine to me. 
Sorry for not opening a pre earlier. Completely forgot.

Thanks a lot.

Also @ElianCodes thanks for the fix-up.

**Comment by HiDeoo** [comment]
No worries at all, altho I still think we do want to link to the new explanation in the Astro Docs as @ElianCodes suggested so I'm gonna reopen this PR to double check that.

**Review by delucis — APPROVED** [review]
Thanks everyone for the work on this PR — I think this looks good now 🙌

**Pull request merged**